uniquify the docstring of unique(itr)

rfourquet opened the pull request:
The docs for `unique(itr)` seems to have been redundant, cf. https://github.com/JuliaLang/julia/pull/15741#issuecomment-308471930.

ararslan commented on base/multidimensional.jl:
```diff
@@ -1298,9 +1298,16 @@ hash(x::Prehashed) = x.hash
 
 Returns an array containing only the unique elements of the iterable `itr`, in
 the order that the first of each set of equivalent elements originally appears.
+The unicity is determined by [`isequal`](@ref).
```
I think "unicity" is a fairly uncommon word. (Indeed, Firefox spellcheck is marking it as incorrect, though a Google search shows it exists.) I think "uniqueness" might be clearer here.

rfourquet replied:
For reference to those interested: https://english.stackexchange.com/questions/225094/uniqueness-vs-unicity

ararslan replied:
I gotta say, I do love the suggestions in one of the answers regarding rhyming. We really need more rhyming docstrings.

nalimilan commented:
See https://github.com/JuliaLang/julia/pull/20800, which I should have merged a few weeks ago.

rfourquet commented:
@nalimilan oups I hadn't search for open PR on that, sorry. I will check if there is something left of this one now that you merged.

rfourquet committed:
uniquify the docstring of unique(itr)

rfourquet commented:
So I still think the docstring were redundant, I updated accordingly. I also updated the example `unique([1; 2; 2; 6])` into `unique([1, 2, 6, 2])` to make it clear that the duplicates don't have to be grouped for `unique` to uniquify them.

nalimilan reviewed (approved):
Indeed, my PR didn't get rid of repetitions.
